app/models/agent_role.rb

```diff
@@ -8,4 +8,9 @@ class AgentRole < ApplicationRecord
   validates :agent, uniqueness: { scope: :organisation, message: "est déjà relié à l'organisation" }
 
   enum access_level: { basic: "basic", admin: "admin" }
+
+  scope :with_export_authorization, -> { where(export_authorization: true) }
+  scope :with_last_name, -> { joins(:agent).where.not(agents: { last_name: nil }) }
+
+  before_save -> { self.export_authorization = true }, if: -> { admin? && export_authorization == false }
```

Comment:
nitpick: ici aussi on pourrait avoir :)
```suggestion
  before_save -> { self.authozried_to_export = true }, if: -> { admin? && !authorized_to_export? }
```